acceptancetests/jujupy/k8s_provider/kubernetes_core.py

```diff
@@ -0,0 +1,332 @@
+                fieldRef:
+                  fieldPath: spec.nodeName
+            - name: PV_DIR
+              value: /mnt/kubernetes
+          volumeMounts:
+            - name: pv-volume
+              mountPath: /mnt/kubernetes
+      volumes:
+        - name: pv-volume
+          hostPath:
+            path: /mnt/kubernetes
+---
+
+# -- Create the standard storage class for running on-node hostpath storage
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  # namespace: kube-system
+  name: {class_name}
+  annotations:
+    storageclass.beta.kubernetes.io/is-default-class: "true"
+  labels:
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: EnsureExists
+provisioner: hostpath
+---
+"""
+
+
+@register_provider
+class KubernetesCore(Base):
+
+    name = K8sProviderType.K8S_CORE
+
+    def __init__(self, client, timeout=1800):
+        super().__init__(client, timeout)
+        self.default_storage_class_name = "juju-storageclass"
+
+    def _ensure_kube_dir(self):
+        # ensure kube credentials
+        self.client.juju('scp', ('kubernetes-master/0:config', self.kube_config_path))
+
+        # ensure kubectl by scp from master
+        self.client.juju('scp', ('kubernetes-master/0:/snap/kubectl/current/kubectl', self.kubectl_path))
+
+    def _ensure_cluster_stack(self):
+        self.__ensure_lxd_profile(
+            LXD_PROFILE,
+            self.client.model_name  # current model now is the default IAAS hosted model for hosting k8s cluster.
+        )
+        self.__deploy_stack_bundle(BUNDLE)
+
+    def _ensure_cluster_config(self):
+        self.__ensure_tmp_fix_for_ingress()
+        self.__ensure_storage_provisoner_default_sc()
+
+    def _node_address_getter(self, node):
+        # TODO(ycliuhw): implement here once described k8s core node to find the corrent node ip.
+        raise NotImplemented()
+
+    def __ensure_lxd_profile(self, profile, model_name):
+        profile = profile.format(model_name=model_name)
+        with subprocess.Popen(('echo', profile), stdout=subprocess.PIPE) as echo:
+            o = subprocess.check_output(
+                ('lxc', 'profile', 'edit', 'juju-%s' % model_name),
+                stdin=echo.stdout
+            ).decode('UTF-8').strip()
+            logger.debug(o)
+
+    def __deploy_stack_bundle(self, bundle_content):
+        with tempfile.NamedTemporaryFile() as f:
+            f.write(bundle_content)
+            self.client.deploy_bundle(f.name, static_bundle=True)
+
+        # Wait for the deployment to finish.
+        self.client.wait_for_started(timeout=self.timeout)
+
+        # wait for cluster to stabilize
+        self.client.wait_for_workloads(timeout=self.timeout)
+
+        # get current status with tabular format for better debugging
+        self.client.juju(self.client._show_status, ('--format', 'tabular'))
+
+    def __ensure_tmp_fix_for_ingress(self):
+        # tmp fix kubernetes core ingress issue
```

Comment:
what's the ingress issue?

Reply:
yeah, this was a fix for kubernetes-core ingress issue(the same issue like in microk8s) - https://github.com/ubuntu/microk8s/issues/222.
At that time, It was fixed in microk8s but not in kubernetes core.